docs/guides-agent-app.md

```diff
@@ -0,0 +1,485 @@
+
+This is because a token has 18 decimals, so 1 unit of a token is actually 0.000000000000000001 tokens. This is not what we want.
+
+In order to mint a full token from the CLI we need to pass the full number, which will then be interpreted with 18 decimals.
+In our case this is a 1 followed by eighteen 0s, or `1000000000000000000`.
+
+Finally, the usual warning: running the above command will trigger a vote in B to create and send a token to A's Agent App: we'll need to vote *yes* to confirm the minting of the token.
+
+You can do this either directly through the UI or by running:
+
+```
+dao exec <name of dao B> <voting app address of dao B> vote 0 true true --environment aragon:rinkeby --apm.ipfs.rpc https://ipfs.eth.aragon.network/ipfs/
+```
+
+On the UI, the vote will look something like this.
+
+<p align="center">
+   <img width="800" src="/docs/assets/agent-guide/agent-8.png">
+</p>
+
+Once you've voted *yes*, switch over to the Token Manager app (click on *Tokens* in the left sidebar).
+
+<p align="center">
+   <img width="800" src="/docs/assets/agent-guide/agent-9.png">
+</p>
+
+You should see that you've successfully added another token holder (your Agent app)!
+
+### **3. Create a vote in B to add a third entity**
+
+As in step 2, we'll run `dao exec` again, except this time the first argument to `mint` will be the address of the third entity we want to add to B.
+
+```
+dao exec <name of dao B> <token manager app address of dao B> mint <third entity's address> 1000000000000000000 --environment aragon:rinkeby --apm.ipfs.rpc https://ipfs.eth.aragon.network/ipfs/
+```
+
+Running the above will create an open vote in B. Again we'll need to vote *yes* to confirm the minting.
+
+<p align="center">
+   <img width="800" src="/docs/assets/agent-guide/agent-10b.png">
+</p>
+
+As before, you can either do this through the UI or run the same command we ran at the end of step 2 with one small modification: 
+
+This time the first argument to `vote` will be a `1` and not a `0`, since the id of this new vote is 1. Remember that vote ids start at zero and increment by one each time a vote is created.
+
+If you go to the Voting app after you've voted, you'll see that the *yes* vote is only 50%.
+
+<p align="center">
+   <img width="800" src="/docs/assets/agent-guide/agent-10.png">
+</p>
+
+That's because only 1 / 2 tokenholders have voted. The other tokenholder (A's Agent app) still has to vote.
+
+
+### **4. Use A's Agent app to take part in B's vote**
+
+In order to close and enact the vote, we'll use A's Agent app to vote yes to adding a third entity to B.
+
+But before we do this, we need to introduce the `dao act` command.
+
+According to the documentation:
+
+>`dao act` provides some syntax sugar over `dao exec` for executing actions using Agent app instances in a DAO.
+
+In plain english, `dao act` acts like `dao exec` except the first argument to `dao act` is an agent app address rather than a DAO address.
```

Comment:
Not only this, but `dao act` can be used to interact with external contracts and not only with apps within the DAO (which is the only thing that `dao exec` does)